src/App.tsx

```diff
@@ -1,223 +1,293 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+// import { TodoInfo } from './components/TodoInfo/TodoInfo';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorTypes, FilterTypes } from './types/Enums';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isError, setIsError] = useState(false);
+  const [filterType, setFilterType] = useState('All');
+  // const [todoTitle, setTodoTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodosIds, setDeletingTodosIds] = useState<number[]>([]);
+  const [isToggledAll, setIsToggledAll] = useState(false);
+  const [todosToUpdate, setTodosToUpdate] = useState<Todo[]>([]);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    if (user) {
+      getTodos(user.id)
+        .then(setTodos)
+        .catch(() => {
+          setIsError(true);
+          setErrorMessage(ErrorTypes.UnableToLoad);
+        });
+    }
+
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+  }, [user]);
+
+  const handleButtonClickAll = () => {
+    setFilterType(FilterTypes.All);
+  };
+
+  const handleButtonClickActive = () => {
+    setFilterType(FilterTypes.ACTIVE);
+  };
+
+  const handleButtonClickCompleted = () => {
+    setFilterType(FilterTypes.COMPLETED);
+  };
```

Comment:
Could we get rid of these functions and pass into footer only setFilterType functions?